README.md

```diff
@@ -6,7 +6,80 @@
+distribute computations to its provider nodes and allow requestors to utilize its unique 
+potential - which can lie in its combined computing power, the geographical distribution 
+or its censorship resistance.
+
+## Golem's requestor setup
+
+Golem's requestor-side configuration consists of two separate components:
+* the [`yagna` daemon](https://github.com/golemfactory/yagna) - your node in the 
+  new Golem network, responsible for communication with the other nodes, running the 
+  market and providing easy access to the payment mechanisms.
+* the requestor agent - the part that the developer of the specific Golem application
+  is responsible for.
+
+The daemon and the requestor agent communicate using three REST APIs which 
+`yapapi` - Golem's Python high-level API - aims to abstract to large extent to make 
+application development on Golem as easy as possible.
+
+## How to use this API?
+
+Assuming you have your Golem node up and running (you can find instructions on how to 
+do that in the [yagna repository](https://github.com/golemfactory/yagna) and in our 
+[handbook](https://handbook.golem.network)), what you need to do is:
+* *prepare your payload* - this needs to be a Docker image containing your application
+  that will be executed on the provider's end. This image needs to have its volumes
+  mapped in a way that will allow the supervisor module to exchange data (write and 
+  read files) with it. This image needs to be packed and uploaded into golem's image repository 
+  using or dedicated tool - [`gvmkit-build`](https://pypi.org/project/gvmkit-build/). 
+* *create your requestor agent* - this is where `yapapi` comes in. Utilizing our high-level
+  API, the creation of a requestor agent should be straighforward and require minimal effort.
+  You can use examples contained in this repository (blender and hashcat) as references.
+
+### Components
+
+There are a few components that are crucial for any requestor agent app:
+
+#### Engine
+
+The heart of the high-level API is the requestor's task runner engine (`yapapi.runner.Engine`).
+You tell it, among others, which package (VM image) will be used to run your tasks, 
+how much you'd like to pay and how many providers you'd like to involve in the execution.
+Finally, you feed it the worker script and a list of `Task` objects to execute on providers. 
+
+#### Worker script
+
+The `worker` will most likely be the very core of your requestor app. You need to define
+this function in your agent code and then you pass it to the runner Engine.
+
+It receives the a `WorkContext` (`yapapi.runner.ctx.WorkContext`) object that serves 
+as an interface between your script and the execution unit within the provider. 
+Using the work context, you define the steps that the provider needs to execute in order
+to complete the job you're giving them.
+
+Depending on the number of workers, and thus, the providers that your runner Engine utilizes,
```

Comment:
This suggests that the number of workers is equal to the number of providers. In fact, the number of providers is bounded by the number of workers.

Reply:
@azawlocki maybe if I change it to "the maximum number of providers that your runner Engine utilizes... " ?

Reply:
"... concurrently/in parallel"

Reply:
👍